src/nupic/engine/RegionImplFactory.cpp

```diff
@@ -211,13 +219,34 @@ namespace nupic
 RegionImplFactory & RegionImplFactory::getInstance()
 {
   static RegionImplFactory instance;
-  if (cpp_packages.empty())
+  // Create C++ regions
+  if (cpp_regions.empty())
   {
-    cpp_packages["TestNode"] = new RegisteredRegionImpl<TestNode>();
-    cpp_packages["VectorFileEffector"] = new RegisteredRegionImpl<VectorFileEffector>();
-    cpp_packages["VectorFileSensor"] = new RegisteredRegionImpl<VectorFileSensor>();
+    cpp_regions["TestNode"] = new RegisteredRegionImpl<TestNode>();
+    cpp_regions["VectorFileEffector"] = new RegisteredRegionImpl<VectorFileEffector>();
+    cpp_regions["VectorFileSensor"] = new RegisteredRegionImpl<VectorFileSensor>();
   }
 
+  // Create Python regions
+  // nupic.regions
+  RegionImplFactory::registerPyRegion("nupic.regions.AnomalyRegion", "AnomalyRegion");
+  RegionImplFactory::registerPyRegion("nupic.regions.CLAClassifierRegion", "CLAClassifierRegion");
+  RegionImplFactory::registerPyRegion("nupic.regions.ImageSensor", "ImageSensor");
+  RegionImplFactory::registerPyRegion("nupic.regions.KNNAnomalyClassifierRegion", "KNNAnomalyClassifierRegion");
+  RegionImplFactory::registerPyRegion("nupic.regions.KNNClassifierRegion", "KNNClassifierRegion");
+  RegionImplFactory::registerPyRegion("nupic.regions.PCANode", "PCANode");
+  RegionImplFactory::registerPyRegion("nupic.regions.PyRegion", "PyRegion");
+  RegionImplFactory::registerPyRegion("nupic.regions.RecordSensor", "RecordSensor");
+  RegionImplFactory::registerPyRegion("nupic.regions.SPRegion", "SPRegion");
+  RegionImplFactory::registerPyRegion("nupic.regions.SVMClassifierNode", "SVMClassifierNode");
+  RegionImplFactory::registerPyRegion("nupic.regions.TPRegion", "TPRegion");
+  RegionImplFactory::registerPyRegion("nupic.regions.TestNode", "TestNode");
+  RegionImplFactory::registerPyRegion("nupic.regions.TestRegion", "TestRegion");
+  RegionImplFactory::registerPyRegion("nupic.regions.UnimportableNode", "UnimportableNode");
+
+  // nupic.regions.extra
+  RegionImplFactory::registerPyRegion("nupic.regions.GaborNode2", "GaborNode2");
```

Comment:
should this be `nupic.regions.extra.GaborNode2`? Maybe we can leave these out of the defaults for now. Also, perhaps the default Python regions could be added in `nupic/engine/__init__.py` rather than in the C++? C++ clients would need to explicitly add them but we wouldn't have to keep this list in sync with the nupic codebase.